soc/xtensa/intel_adsp/common/bootloader/boot_entry.S

```diff
@@ -14,10 +14,12 @@
  */
 
 #include <soc/shim.h>
-#include <soc/platform.h>
 #include <soc/memory.h>
 #include <xtensa/corebits.h>
 #include <xtensa/config/core-isa.h>
+#include "platform.h"
+
+#define PLATFORM_PRIMARY_CORE_ID 0
```

Comment:
I don't know how this happens, but with this PR, 

```
sof/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/bootloader/boot_entry.S: Assembler messages:
sof/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/bootloader/boot_entry.S:58: Error: missing ')'
sof/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/bootloader/boot_entry.S:58: Error: junk at end of line, first unrecognized character is `U'
xt-xcc ERROR:  /xtensa/XtDevTools/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-as returned non-zero status 1
```

Looking at preprocessor output for boot_entry.S, it seems the xcc-as doesn't like Zephyr BIT(1) definition in assembly files.

```
     58 wnd0_base_val:
     59  .word (0xbe000000 + 0x4000) | (1UL << (1)) | (1UL << (0))
```

Reply:
Heh, nice synchronicity.  I bumped into exactly this error (though possibly in a different place) building the Zephyr thread_apis test with xcc testing the (unrelated) linker fix yesterday.  I haven't gotten back to investigate.  But yeah, apparently there's something glitchy here.  Maybe the assembler doesn't like ANSI "UL" suffixes?  Surely it supports bit shifting...

Reply:
Works now in lates version.